modules/cpmso-feat-auto-update.adoc

```diff
@@ -2,12 +2,49 @@
+ifndef::openshift-dedicated,openshift-rosa[]
+The `RollingUpdate` update strategy automatically propagates changes to your control plane configuration.
+This update strategy is the default configuration for the control plane machine set.
+endif::openshift-dedicated,openshift-rosa[]
+//For ROSA:
+ifdef::openshift-rosa[]
+On ROSA (classic architecture) clusters, control plane machine sets automatically propagate changes to your control plane configuration.
+The Control Plane Machine Set Operator creates a replacement control plane machine with the configuration that is specified in the `ControlPlaneMachineSet` CR. The Operator replaces control plane machines in a way that mitigates any potential negative effects on cluster API or workload availability.
+endif::openshift-rosa[]
+//For OSD:
+ifdef::openshift-dedicated[]
+On {product-title} clusters, the control plane machine sets automatically propagate changes to your control plane configuration.
+The Control Plane Machine Set Operator creates a replacement control plane machine with the configuration that is specified in the `ControlPlaneMachineSet` CR. The Operator replaces control plane machines in a way that mitigates any potential negative effects on cluster API or workload availability.
+endif::openshift-dedicated[]
+//Not for ROSA/OSD:
+ifndef::openshift-dedicated,openshift-rosa[]
+
+For clusters that use the `RollingUpdate` update strategy, the Operator creates a replacement control plane machine with the configuration that is specified in the CR.
+endif::openshift-dedicated,openshift-rosa[]
+//Anything not enclosed in an ifdef or ifndef prints for everyone:
+
+When the replacement control plane machine is ready, the Operator deletes the control plane machine that is marked for replacement because the machine is either out of spec or in an unhealthy state.
+The replacement machine then joins the control plane.
 
-The `RollingUpdate` update strategy automatically propagates changes to your control plane configuration. This update strategy is the default configuration for the control plane machine set.
+If multiple control plane machines are marked for replacement, the Operator protects etcd health during replacement by repeating this replacement process one machine at a time until it has replaced each machine.
 
-For clusters that use the `RollingUpdate` update strategy, the Operator creates a replacement control plane machine with the configuration that is specified in the CR. When the replacement control plane machine is ready, the Operator deletes the control plane machine that is marked for replacement. The replacement machine then joins the control plane.
+ifdef::openshift-dedicated,openshift-rosa[]
+[NOTE]
+====
+You cannot request that control plane machine replacements happen only during maintenance windows. The Control Plane Machine Set Operator acts to ensure cluster stability. Waiting for a maintenance window could result in cluster stability being compromised.
```

Comment:
This is great; thanks for highlighting this

Reply:
big plus one haha ;)